data/src/main/java/com/linkedin/data/DataMap.java

```diff
@@ -447,5 +554,6 @@ private void instrumentAccess(Object key)
   private boolean _madeReadOnly = false;
   private boolean _instrumented = false;
   private Map<String, Integer> _accessMap;
+  private SpecificMap _specificMap;
```

Comment:
I think a separate _specificMap is confusing - you cannot have both the _mapp field in superclass and _specificMap set to non-null values, right? The new field takes up some extra memory. Ideally, we need a deeper change that would make _map be able to accommodate a SpecificMap instance.

Reply:
This has been addressed in the latest diff. I moved it to a boolean encapsulatesSpecificMap that is set on init. This is faster than doing an instanceof check at runtime. Also this is safe because the map never changes after being assigned.